nikola/plugins/compile/rest/post_list.py

```diff
@@ -246,18 +251,16 @@ def _do_post_list(start=None, stop=None, reverse=False, tags=None, categories=No
     if sections:
         timeline = [p for p in timeline if p.section_name(lang).lower() in sections]
 
-    for post in timeline:
-        if tags:
-            cont = True
-            tags_lower = [t.lower() for t in post.tags]
-            for tag in tags:
-                if tag in tags_lower:
-                    cont = False
-
-            if cont:
-                continue
-
-        filtered_timeline.append(post)
+    if tags:
+        tags = {t.strip().lower() for t in tags.split(',')}
+        if require_all_tags:
+            compare = lambda x, y: x.issubset(y)
+        else:
+            compare = lambda x, y: x & y
+        for post in timeline:
+            post_tags = {t.lower() for t in post.tags}
+            if compare(tags, post_tags):
+                filtered_timeline.append(post)
```

Comment:
This is already in an if tags: conditional; that case will never actually trigger, or I cant see how it will trigger (now that i see it in more context)